ext/drx/drx.c

```diff
@@ -2787,27 +2867,31 @@ drx_event_restore_state(void *drcontext, bool restore_memory,
         /* Nothing to do if nobody had never called expand_scatter_gather() before. */
         return true;
     }
+    if (!info->fragment_info.app_code_consistent) {
+        /* Can't verify application code.
+         * XXX i#2985: is it better to keep searching?
+         */
+        return true;
+    }
     instr_init(drcontext, &inst);
     byte *pc = decode(drcontext, dr_fragment_app_pc(info->fragment_info.tag), &inst);
     if (pc != NULL) {
+        scatter_gather_info_t sg_info;
+        get_scatter_gather_info(&inst, &sg_info);
         if (instr_is_gather(&inst)) {
-            if (!info->fragment_info.app_code_consistent) {
-                /* Can't verify application code.
-                 * XXX i#2985: is it better to keep searching?
-                 */
-                return true;
-            }
-            scatter_gather_info_t sg_info;
-            get_scatter_gather_info(&inst, &sg_info);
             if (sg_info.is_evex) {
                 success = success &&
                     drx_try_to_detect_avx512_gather_sequence(drcontext, info, &inst,
```

Comment:
Is there a more descriptive name: this name sounds like it's non-mutating, when in fact it is modifying state.  How about "drx_restore_state_for_avx512_gather()" or sthg.

Reply:
Done.